File: crates/weave-graph-parse/tests/fixtures/sample.py
from collections import OrderedDict


class Greeter:
    def __init__(self, name):
        self.name = name

    def greet(self):
        return format_name(self.name)


class LoudGreeter(Greeter):
    def greet(self):
        return self.greet().upper()


def format_name(name):
    return name.strip()

File: crates/weave-graph-parse/tests/fixtures/sample.ts
import { pad } from "./strings";

interface Named {
    getName(): string;
}

class Greeter implements Named {
    constructor(private name: string) {}

    getName(): string {
        return this.name;
    }

    greet(): string {
        return formatName(this.getName());
    }
}

function formatName(name: string): string {
    return pad(name);
}
